sample-3.md

```diff
@@ -0,0 +1,548 @@
+but rather than using a single YAML, we will apply each piece one step at a time, 
+explaining what it does as we do so.
+
+We will create the following topology:
+
+Image from 2.
+
+- [`Pod` named `curl`](#event-producer), acting as the event producer.
+- `Broker` named `default`.
+- Two [`Trigger`s](#triggers), each with a mutually exclusivefilter pointing at distinct 
+  subscribers.
+- Two Kubernetes [`Service`s](#event-consumers), each pointing at a unique:
+    - [`Deployment`](#event-consumers), acting as the event consumer.
+
+## Helpers
+
+First we will create useful shell variables to allow us to not have to type 
+things out every time.
+
+```shell
+K_NAMESPACE=kn-eventing-step-by-step-sample
+```
+
+## Namespace
+
+Create the namespace.
+
+```shell
+kubectl create namespace $K_NAMESPACE
+```
+
+Now that the namespace has been created, we will label it to get Knative eventing
+setup. Adding this label will:
+- Create `ServiceAccount`s.
+- Create `RoleBinding`s to give permissions to those `ServiceAccount`s.
+- Create a `Broker` named `default`.
+
+```shell
+kubectl label namespace $K_NAMESPACE knative-eventing-injection=enabled
+```
+
+### Broker
+
+What is a `Broker`? It can be thought of as an event mesh. Event
+producers send events into the `Broker`. Event consumers register the kinds of events
+they are interested in with the `Broker` (via `Trigger`s, 
+[described later](#triggers)). The
+`Broker` ensures that every event sent to the `Broker` by event producers is sent
+to all interested event consumers. To learn more about `Broker`s, see [`Broker` in
+Eventing Concepts](todo).
+
+
+Let's verify that the `Broker` is in a healthy state.
+
+```shell
+kubectl -n $K_NAMESPACE get broker default
+```
+
+This should return something like:
+
+```shell
+NAME      READY   REASON   HOSTNAME                                                           AGE
+default   True             default-broker.kn-eventing-step-by-step-sample.svc.cluster.local   1m
+```
+
+We are particularly interested in the second column, `READY`. We want that value to
+be `True`. It may take a minute or two to become `True`. If it is not `True` within
+two minutes, then something has gone wrong. Hopefully the `REASON` column gives a
+useful error message. See our [Broker Debugging Guide](somewhere) to help fix the
+error.
+
+## Event Consumers
+
+The first thing we are going to do is create our event consumers. These are 
+`Deployment`s, and `Service`s that point at them. We are creating two distinct
+event consumers, so that we can show how to selectively send events to distinct
+event consumers later.
+
+Setup the '-dev' `Service` and `Deployment`.
+
+**TODO** Use a Google Cloud style tabbed interface to allow people to choose if they
+wnat bash, or download a yaml and run, something else... Similar to Google Cloud
+pages that show gcloud, Cloud Console, and API tabs for their examples.
+
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
```

Comment:
```suggestion
kubectl -n $K_NAMESPACE apply -f - <<END
```

Reply:
Done.

I'm choosing not to accept the commits, because Prow sometimes freaks out about the CLA if multiple commit authors are in a PR.